x-pack/plugins/alerting/server/rules_client/methods/bulk_edit.ts

```diff
@@ -540,7 +540,20 @@ async function getUpdatedAttributesFromOperations(
     // the `isAttributesUpdateSkipped` flag to false.
     switch (operation.field) {
       case 'actions': {
-        await validateActions(context, ruleType, { ...attributes, actions: operation.value });
+        try {
+          await validateActions(context, ruleType, {
+            ...attributes,
+            actions: operation.value,
+          });
+        } catch (e) {
+          // If validateActions fails on the first attempt, it may be because of legacy rule-level frequency params
+          attributes = await attemptToMigrateLegacyFrequency(
```

Comment:
I have few questions here:
- if  we keep the logic the way it is, do we need to re-validate after `attemptToMigrateLegacyFrequency`
- Also I am confuse around the rule level and action frequency, I thought that we will check if we have the frequency on the action then we will use this one and if we do not have it we will the rule level frequency

Reply:
I moved the re-validation step into `attemptToMigrateLegacyFrequency` in response to @maximpn's feedback: https://github.com/elastic/kibana/pull/145637#discussion_r1058906340

>  if we have the frequency on the action then we will use this one and if we do not have it we will the rule level frequency

Part of the backend spec was to throw an error if both rule-level and action-level frequencies were present. The task runner will still let action-level frequency supersede rule-level frequency as a fallback behavior, in case that situation somehow sneaks by, but we want to at least try to prevent it.